angr/knowledge_plugins/functions/soot_function.py

```diff
@@ -0,0 +1,963 @@
+
+import os
+import logging
+import networkx
+import string
+import itertools
+from collections import defaultdict
+from itanium_demangler import parse
+
+import claripy
+from ...errors import SimEngineError, SimMemoryError
+from ...procedures import SIM_LIBRARIES
+
+l = logging.getLogger(name=__name__)
+
+
+class SootFunction(object):
```

Comment:
`(object)` is unnecessary in Python 3.

Reply:
Also, it seems to me that `SootFunction` is sharing *a lot of code* with `Function`. Maybe we should have a base class for those two classes?

Reply:
True. I think Andrea forgot to refactor that part. Refactored now.